zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/clock/DbClockModification.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.clock;
+
+import io.camunda.zeebe.db.DbValue;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.EnumProperty;
+import io.camunda.zeebe.msgpack.property.ObjectProperty;
+import io.camunda.zeebe.protocol.impl.record.value.clock.ClockRecord;
+import io.camunda.zeebe.stream.api.StreamClock.ControllableStreamClock.Modification;
+import java.time.Duration;
+import java.time.Instant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class DbClockModification extends UnpackedObject implements DbValue {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DbClockModification.class);
+
+  private final EnumProperty<ModificationType> typeProperty =
+      new EnumProperty<>("type", ModificationType.class, ModificationType.NONE);
+  private final ObjectProperty<ClockRecord> modificationProperty =
+      new ObjectProperty<>("modification", new ClockRecord());
```

Comment:
:upside_down_face: Could also just be a `LongProperty`, that way you don't use a record value in the state

Reply:
Now that we merged both properties in the previous PR, yes :smile: